src/errors/BaseError.ts

```diff
@@ -7,16 +7,35 @@ class BaseIsomerError extends Error implements IsomerInternalError {
 
   message: string
 
-  code = "BaseError"
+  code = "BaseError" // name of the error
 
   isIsomerError = true
 
-  constructor(status = 500, message = "Something went wrong") {
+  meta: Record<string, unknown> // additional properties that provides context for the error
+
+  isV2Err = false // indicates if this is the new error format
+
+  componentCode: string // a selection from ComponentTypes
+
+  fileCode: string // a selection from FileCodes
+
+  constructor({
+    status = 500,
+    code = "BaseError",
+    message = "Something went wrong",
+    meta = {},
+    componentCode = ComponentTypes.Other,
+    fileCode = "000",
+  }) {
     super()
     Error.captureStackTrace(this, this.constructor)
     this.name = this.constructor.name
+    this.code = code
```

Comment:
fyi, in the downstream, this was changed to

`this.code = this.constructor.name` - seems to be identical to what we have at present

Reply:
👍🏼